Daniel_Heffner/angular/product_factory/scriptular.js

```diff
@@ -0,0 +1,76 @@
+var appModule = angular.module('product_app', []);
+
+appModule.factory('product_factory', function(){
+	var products = [
+		{name: 'Sample Product', price: 39.95},
+	];
+	var factory = {}
+	factory.getProducts = function(callback){
+		callback(products);
+		return false;
+	}
+	factory.addProduct = function(name, price){
+		if (name != '' && (price != '' && price > 0)){
+			products.push({name: name, price: price});
+			var errors = {success: true};
+			return errors;
+		} else {
+			var errors = {success: false};
+			if (name == '') errors.name = 'Products must have names!';
+			if (price == '' || !(price > 0)) errors.price = 'Products must cost something!';
+			return errors;
+		}
+	}
+	factory.destroyProduct = function(idx){
```

Comment:
In a real-world application, you'll have an $.ajax call here. You won't actually delete the data from `products` until the ajax completes, after which you would call a callback.